### PR TITLE
Allow Friends Into Locked Lobby

### PR DESCRIPTION
**Pull request by Rxann**

### CODE REVIEW

File: src/hooks/player_management/assign_physical_index.cpp [hunk 1]
```diff
@@ -128,9 +128,18 @@ namespace big
 
 					if (g.session.lock_session && g_player_service->get_self()->is_host())
 					{
-						dynamic_cast<player_command*>(command::get(RAGE_JOAAT("breakup")))->call(plyr, {});
-						g_notification_service->push_warning("Lock Session",
-						    std::format("A player with the name of {} has been denied entry", plyr->get_net_data()->m_name));
+						if (plyr.is_friend() && g.session.allow_friends_into_locked_session)
+						{
+							g_notification_service->push_success("Lock Session",
+							    std::format("A player friended player with the name of {} has been allowed to join the locked lobby!",
```
Comment (tupoy-ya) [inline src/hooks/player_management/assign_physical_index.cpp]: We call it a session and don't use `!` anywhere else so we probably shouldn't here.

**Review by Yimura — DISMISSED** [review]

**Comment by gir489returns** [comment]
This is failing because of code reasons, not because of heap exhaustion.

D:\a\YimMenu\YimMenu\src\hooks\player_management\assign_physical_index.cpp(131): error C2039: 'is_friend': is not a member of 'std::shared_ptr<big::player>'

**Comment by Rxann** [comment]
That should fix it

File: src/hooks/player_management/assign_physical_index.cpp [hunk 2]
```diff
@@ -128,9 +128,18 @@ namespace big
 
 					if (g.session.lock_session && g_player_service->get_self()->is_host())
 					{
-						dynamic_cast<player_command*>(command::get(RAGE_JOAAT("breakup")))->call(plyr, {});
-						g_notification_service->push_warning("Lock Session",
-						    std::format("A player with the name of {} has been denied entry", plyr->get_net_data()->m_name));
+						if (plyr->is_friend() && g.session.allow_friends_into_locked_session)
+						{
+							g_notification_service->push_success("Lock Session",
+							    std::format("A player friended player with the name of {} has been allowed to join the locked session",
```
Comment (BugisoftRSG) [inline src/hooks/player_management/assign_physical_index.cpp]: "A player friended player" -> "A friend"

Comment (Rxann) [reply src/hooks/player_management/assign_physical_index.cpp]: Good catch

**Review by Yimura — APPROVED** [review]

**Pull request merged**